backEnd/src/auth/auth.ts

```diff
@@ -4,26 +4,23 @@ import cookie from 'cookie'
 import prisma from '../prismaClient'
 import { NextFunction, Request, Response } from 'express'
 
-const createNewToken = (user: any) : string => {
+const createNewToken = (user: any): string => {
     return jwt.sign(
-            {id: user.id, role: user.role, username: user.username}, 
-            process.env.JWT_SECRET as jwt.Secret, 
-            { expiresIn: process.env.JWT_EXPIRATION })
+        { id: user.id, role: user.role, username: user.username },
+        process.env.JWT_SECRET as jwt.Secret,
+        { expiresIn: process.env.JWT_EXPIRATION })
 }
 
 
-export const validEmployee = async (req: Request, res: Response, next: NextFunction) => {
-    const decoded = jwt.decode(req.cookies.ORDER_UP_TOKEN, {complete: true})
+export const validEmployee = (rolesToCheck: string | Array<string> = ["ADMIN", "MANAGER"]) => async (req: Request, res: Response, next: NextFunction) => {
```

Comment:
Just want to check, this is TypeScript functionality right?

Reply:
Hey, most of this is just standard JS.
```tsx
export const validEmployee = (rolesToCheck: string | Array<string> = ["ADMIN", "MANAGER"]) =>
```
The TypeScript portions are here:
```tsx
: string | Array<string>
```
These just specify the types that I expect to be passed to this function, either a string, or an array of strings.
The default value `["ADMIN", "MANAGER"]` is standard JS, and I am using currying to create a function which checks for specific roles you pass in.
An example of usage is this (also seen in `menuItem.controller.ts`):
 ```tsx
validEmployee(["MANAGER","ADMIN"])
```
This creates an express function that checks if the employee is a manager or an admin.